blockmanager.go

```diff
@@ -480,7 +488,6 @@ func (b *blockManager) handleTxMsg(tmsg *txMsg) {
 	// instances of trying to fetch it, or we failed to insert and thus
 	// we'll retry next time we get an inv.
 	txHash := tmsg.tx.Sha()
-	delete(tmsg.peer.requestedTxns, *txHash)
```

Comment:
We're using `peerInfo.requestedTxns` and holding all peers in `blockManager.peers` so we can easily delete the tx from the peer requests queue.